chapters/Syntax.md

```diff
@@ -1,36 +1,225 @@
+
+The ABNF for a **did:btc1** identifier is as follows:
 
 ```abnf
-did-btc1 = "did:btc1:" [ version ":" ] [ network ":" ] id-bech32
-version	= 1*DIGIT
-network =  "mainnet" / "signet" / "testnet" / "regtest"
-id-bech32 = key-value / hash-value
-hash-value = "x1" bech32-encoding
-key-value = "k1" bech32-encoding
+did-btc1 = "did:btc1:" id-bech32
+id-bech32 = key-encoding / hash-encoding
+hash-encoding = "x1" bech32-encoding
+key-encoding = "k1" bech32-encoding
 bech32-encoding = *bech32char
 bech32char = "0" / "2" / "3" / "4" / "5" / "6" / "7" / "8" / "9" / "a" / "c" / 
 "d" / "e" / "f" / "g" / "h" / "j" / "k" / "l" / "m" / "n" / "p" / "q" / "r" / 
 "s" / "t" / "u" / "v" / "w" / "x" / "y" / "z" 
 ```
 
-ABNF is defined by the [IETF RFC5234](https://datatracker.ietf.org/doc/html/rfc5234).
+ABNF is defined by [IETF RFC5234](https://datatracker.ietf.org/doc/html/rfc5234).
+
+### Version Interpretation
+
+The purpose of the `version` is to identify incompatible changes made in the
+specification (e.g., a change to the way ::Beacon signals:: are constructed and
+interpreted). The updated specification may also change the way that the
+**did:btc1** identifier is encoded.
+
+There are two consequences. The first is that the `version` MUST be able to go
+beyond the domain of a nibble (1-16). To support this, the start of the decoded
+`id-bech32` MUST be interpreted as follows:
+
+1. Set `version` to `1`.
+1. Start with the first nibble (the higher nibble of the first byte).
+1. Add the value of the current nibble to `version`.
+1. If the value of the nibble is hexadecimal `F` (decimal `15`), advance to the
+   next nibble (the lower nibble of the current byte or the higher nibble of the
+   next byte) and return to the previous step.
+
+More simply:
+
+```text
+version = 1 + sum of all nibbles up to and including first non-hexadecimal-F
+```
+
+What appears beyond the version is version-specific. Implementations MUST NOT
+attempt to interpret **did:btc1** identifiers with an unknown `version`.
+
+For illustration purposes only, assume that the nibble following the version
+always represents the network and that the allowable values for the network
+don't change. The interpretation of the`version` and `network` would be as
+follows (spaces between bytes added for readability):
+
+| Decoded bytes     | Version | Network         |
+|-------------------|--------:|-----------------|
+| *00* 03 c7 ...    |       1 | bitcoin         |
+| *23* 03 c7 ...    |       3 | testnet v3      |
+| *C9* 03 c7 ...    |      13 | custom signet 2 |
+| *F2 90* 03 c7 ... |      18 | custom signet 2 |
+| *FF 72* 03 c7 ... |      38 | regtest         |
+
+### Encoding a **did:btc1** Identifier
+
+Given:
+
+* `idType` - one of:
+  * "key"
+  * "hash"
+* `version` - number
+* `network` - one of:
+  * "bitcoin"
+  * "signet"
+  * "regtest"
+  * "testnet3"
+  * "testnet4"
+  * number
+* `idBytes` - byte array, one of:
```

Comment:
```suggestion
* `genesisBytes` - byte array, one of:
```

I think I would prefer to use this name as consistent with the rest of the spec.

Reply:
Open to a discussion as to by idBytes is better. 

I just see the idBytes as the whole thing. Wheras the genesisBytes are the seed for the ID